tests/registry_test.go

```diff
@@ -0,0 +1,167 @@
+
+			var app model.App
+
+			BeforeEach(func() {
+				app = apps.Create(user, "--no-remote")
+				builds.Create(user, app)
+			})
+
+			AfterEach(func() {
+				apps.Destroy(user, app)
+			})
+
+			Specify("that user can list that app's registry information", func() {
+				sess, err := cmd.Start("deis registry:list --app=%s", &user, app.Name)
+				Eventually(sess).Should(Say("=== %s Registry", app.Name))
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess).Should(Exit(0))
+			})
+
+			Specify("that user cannot unset an invalid registry information", func() {
+				sess, err := cmd.Start("deis registry:unset --app=%s munkafolyamat", &user, app.Name)
+				// TODO: should unsetting a bogus registry info return 0 (success?)
+				Eventually(sess, settings.MaxEventuallyTimeout).Should(Say("=== %s Registry", app.Name))
+				Eventually(sess).ShouldNot(Say(`munkafolyamat\s+yeah`, app.Name))
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess).Should(Exit(0))
+			})
+
+			Specify("that user can set a valid registry information", func() {
+				sess, err := cmd.Start("deis registry:set --app=%s username=bob", &user, app.Name)
+				Eventually(sess, settings.MaxEventuallyTimeout).Should(Say("=== %s Registry", app.Name))
+				Eventually(sess).Should(Say(`username\s+bob`))
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess).Should(Exit(0))
+
+				sess, err = cmd.Start("deis registry:list --app=%s", &user, app.Name)
+				Eventually(sess).Should(Say("=== %s Registry", app.Name))
+				Eventually(sess).Should(Say(`username\s+bob`))
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess).Should(Exit(0))
+			})
+
+			Context("and registry information has already been added to the app", func() {
+
+				BeforeEach(func() {
+					sess, err := cmd.Start("deis registry:set --app=%s username=bob", &user, app.Name)
+					Eventually(sess, settings.MaxEventuallyTimeout).Should(Say("=== %s Registry", app.Name))
+					Eventually(sess).Should(Say(`username\s+bob`))
+					Expect(err).NotTo(HaveOccurred())
+					Eventually(sess).Should(Exit(0))
+				})
+
+				Specify("that user can unset that tag from that app", func() {
+					sess, err := cmd.Start("deis registry:unset --app=%s username", &user, app.Name)
+					Eventually(sess, settings.MaxEventuallyTimeout).Should(Say("=== %s Registry", app.Name))
+					Eventually(sess).ShouldNot(Say(`username\s+bob`))
+					Expect(err).NotTo(HaveOccurred())
+					Eventually(sess).Should(Exit(0))
+
+					sess, err = cmd.Start("deis registry:list --app=%s", &user, app.Name)
+					Eventually(sess).Should(Say("=== %s Registry", app.Name))
+					Eventually(sess).ShouldNot(Say(`username\s+bob`))
+					Eventually(sess).ShouldNot(Say(`munkafolyamat\s+yeah`, app.Name))
+					Expect(err).NotTo(HaveOccurred())
+					Eventually(sess).Should(Exit(0))
+				})
+
+			})
+
+			Context("and an application is deployed using a private registry image", func() {
```

Comment:
Usually there would be a "before" to satisfy the conditions described by the context.